### ISSUE TITLE
Cody: VS Code extension - getCodebaseContext should use ActiveTextEditor's filePath instead of workspaceRoot when vscode.window.onDidChangeActiveTextEditor

### ISSUE THREAD

**Issue by marcleblanc2** [sourcegraph/sourcegraph#52235]
#### Feature request description

TL;DR use the `vscode.window.activeTextEditor.document.uri.fsPath` file path instead of `workspaceRoot`,  when calling     `const gitCommand = spawnSync('git', ['remote', 'get-url', 'origin'], { cwd: workspaceRoot })` in 

This would go a long way towards multi-repo workspaces.

Screenshot to help explain
<img width="2032" alt="image" src="https://github.com/sourcegraph/sourcegraph/assets/7050295/d1f2edb7-26c6-469b-b9ec-52bc478d5128">

On the left, I have a file open from the sourcegraph/handbook repo
On the right, I have a file open from the sourcegraph/sourcegraph repo

When I click in the file on the left, I want the Cody codebase to get updated to the handbook repo
When I click in the file on the right, I want the Cody codebase to get updated to the sourcegraph repo

We could implement a new getActiveTextEditorFilePath(), similar to getWorkspaceRootPath() [here](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@f797d27d8112891fc5238ef76b8ae3639682d627/-/blob/client/cody/src/editor/vscode-editor.ts?L16) for efficiency, so that we're only returning the `vscode.window.activeTextEditor.document.uri.fsPath` value.

Then we could add a couple lines to `getCodebaseContext` to prefer the file path of the active editor, but fall back to the workspace root for situations like, there is no file open.

```go
async function getCodebaseContext(config: Config, rgPath: string, editor: Editor): Promise<CodebaseContext | null> {
    const client = new SourcegraphGraphQLAPIClient(config)

    // Try to get the active text editor's file path
    const gitWorkingDirectory = editor.getActiveTextEditorFilePath()

    // If that fails, then the fall back is the workspace root
    if (!gitWorkingDirectory) {
 
            const gitWorkingDirectory = editor.getWorkspaceRootPath()
 
           if (!gitWorkingDirectory) {
                return null
            }
    }
    const gitCommand = spawnSync('git', ['remote', 'get-url', 'origin'], { cwd: gitWorkingDirectory })
```

- We already listen for the active editor change event `vscode.window.onDidChangeActiveTextEditor` [here](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@f797d27/-/blob/client/cody/src/chat/ChatViewProvider.ts?L138:13)
- which calls `updateCodebaseContext` [here](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@f797d27d8112891fc5238ef76b8ae3639682d627/-/blob/client/cody/src/chat/ChatViewProvider.ts?L392)

- `updateCodebaseContext` calls `getCodebaseContext` [here](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@f797d27d8112891fc5238ef76b8ae3639682d627/-/blob/client/cody/src/chat/ChatViewProvider.ts?L865)
- which calls the `git remote get-url origin` [here](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@f797d27d8112891fc5238ef76b8ae3639682d627/-/blob/client/cody/src/chat/ChatViewProvider.ts?L865), with the `cwd: workspaceRoot`, and we should change  workspaceRoot to the file path of the newly activated text editor.

#### Is your feature request related to a problem? If so, please describe.

Even with the many improvements we've made, configuring `cody.codebase` is still a struggle for our users, so this suggestion would be a huge help in the default case for new users where they haven't configured `cody.codebase` yet, as it would automatically set the context for the user each time they open or switch to a file. 

#### Additional context

Slack thread with @abeatrix [here](https://sourcegraph.slack.com/archives/C04MSD3DP5L/p1684534383574339).

**Comment by dominiccooney** [comment]
Played around with this a bit here:

https://www.loom.com/share/2bd43c6c6b314dc89fa4897c6a4eb374 https://github.com/sourcegraph/sourcegraph/tree/dpc/issue52235

**Comment by jasonhawkharris** [comment]
More tickets that illustrate this issue:
[14107](https://sourcegraph.zendesk.com/agent/tickets/14107)
[14103](https://sourcegraph.zendesk.com/agent/tickets/14103)
